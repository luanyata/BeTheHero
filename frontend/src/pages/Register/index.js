import React from "react";
import { Link } from "react-router-dom";

import { FiArrowLeft } from "react-icons/fi";
import "./style.css";

import logoImg from "../../assets/logo.svg";

function Register() {
  return (
    <div className="register-container">
      <div className="content">
        <section>
          <img src={logoImg} alt="Be The Hero" />
          <h1>Cadastro</h1>
          <p>
            Faça seu cadastro, entre na plataforma e ajude pessoas a encontrarem
            os casos da sua ONG.
          </p>
          <Link to="/" className="back-link">
            <FiArrowLeft size={16} color="#E02041" />
            Não tenho cadastro
          </Link>
        </section>

        <form>
          <input placeholder="Nome da ONG" />
          <input type="email" placeholder="E-mail" />
          <input placeholder="WhatsApp" />
          <div className="input-group">
            <input placeholder="Cidade" />
            <input placeholder="UF" style={{ width: 80 }} />
          </div>

          <button className="button" type="submit">
            Cadastrar
          </button>
        </form>
      </div>
    </div>
  );
}

export default Register;
