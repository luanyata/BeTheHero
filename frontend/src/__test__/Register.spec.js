import React from "react";
import { mount } from "enzyme";
import { MemoryRouter } from "react-router-dom";
import Register from "../pages/Register";

describe("<Register />", () => {
  it("should to render the <Register /> component", () => {
    const wrapper = mount(
      <MemoryRouter>
        <Register />
      </MemoryRouter>
    );
    expect(wrapper.debug()).toMatchSnapshot();
  });
});
