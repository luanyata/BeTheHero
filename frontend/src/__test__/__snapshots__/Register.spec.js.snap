// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Register /> should to render the <Register /> component 1`] = `
"<MemoryRouter>
  <Router history={{...}}>
    <Register>
      <div className=\\"register-container\\">
        <div className=\\"content\\">
          <section>
            <img src=\\"logo.svg\\" alt=\\"Be The Hero\\" />
            <h1>
              Cadastro
            </h1>
            <p>
              Faça seu cadastro, entre na plataforma e ajude pessoas a encontrarem os casos da sua ONG.
            </p>
            <Link to=\\"/\\" className=\\"back-link\\">
              <LinkAnchor className=\\"back-link\\" href=\\"/\\" navigate={[Function: navigate]}>
                <a className=\\"back-link\\" href=\\"/\\" onClick={[Function: onClick]}>
                  <FiArrowLeft size={16} color=\\"#E02041\\">
                    <IconBase attr={{...}} size={16} color=\\"#E02041\\">
                      <svg stroke=\\"currentColor\\" fill=\\"none\\" strokeWidth=\\"2\\" viewBox=\\"0 0 24 24\\" strokeLinecap=\\"round\\" strokeLinejoin=\\"round\\" size={16} color=\\"#E02041\\" className={[undefined]} style={{...}} height={16} width={16} xmlns=\\"http://www.w3.org/2000/svg\\">
                        <line x1=\\"19\\" y1=\\"12\\" x2=\\"5\\" y2=\\"12\\" />
                        <polyline points=\\"12 19 5 12 12 5\\" />
                      </svg>
                    </IconBase>
                  </FiArrowLeft>
                  Não tenho cadastro
                </a>
              </LinkAnchor>
            </Link>
          </section>
          <form onSubmit={[Function: handleRegister]}>
            <input placeholder=\\"Nome da ONG\\" value=\\"\\" onChange={[Function: onChange]} />
            <input placeholder=\\"Login\\" value=\\"\\" onChange={[Function: onChange]} />
            <input type=\\"email\\" placeholder=\\"E-mail\\" value=\\"\\" onChange={[Function: onChange]} />
            <input placeholder=\\"Senha\\" type=\\"password\\" value=\\"\\" onChange={[Function: onChange]} />
            <input placeholder=\\"WhatsApp\\" value=\\"\\" onChange={[Function: onChange]} />
            <div className=\\"input-group\\">
              <input placeholder=\\"Cidade\\" value=\\"\\" onChange={[Function: onChange]} />
              <input placeholder=\\"UF\\" style={{...}} value=\\"\\" onChange={[Function: onChange]} />
            </div>
            <button className=\\"button\\" type=\\"submit\\">
              Cadastrar
            </button>
          </form>
        </div>
      </div>
    </Register>
  </Router>
</MemoryRouter>"
`;
